File: src/main/java/ar/edu/um/tesoreria/rest/extern/model/CarreraFacultad.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.extern.model;

import java.io.Serializable;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CarreraFacultad implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -6104456910693552875L;

	private Integer facultadId;
	private Integer planId;
	private Integer carreraId;
	private Long uniqueId;
	private String nombre;
	private String iniciales;
	private String titulo;
	private Byte trabajoFinal;
	private String resolucion;
	private Byte chequeraUnica;
	private Integer bloqueId;
	private Integer obligatorias;
	private Integer optativas;
	private Byte vigente;

	public String getCarreraKey() {
		return this.facultadId + "." + this.planId + "." + this.carreraId;
	}

}

File: src/main/java/um/tesoreria/core/extern/model/kotlin/CarreraFacultad.kt
package um.tesoreria.core.extern.model.kotlin

data class CarreraFacultad(

    var facultadId: Int? = null,
    var planId: Int? = null,
    var carreraId: Int? = null,
    var uniqueId: Long? = null,
    var nombre: String? = null,
    var iniciales: String? = null,
    var titulo: String? = null,
    var trabajoFinal: Byte? = null,
    var resolucion: String? = null,
    var chequeraUnica: Byte? = null,
    var bloqueId: Int? = null,
    var obligatorias: Int? = null,
    var optativas: Int? = null,
    var vigente: Byte? = null,

    ) {

    fun getCarreraKey(): String {
        return this.facultadId.toString() + "." + this.planId + "." + this.carreraId
    }
}
